src/encoded/static/components/search.js

```diff
@@ -552,52 +554,128 @@ const DatasetComponent = (props, reactContext) => {
-        }
 
         // Dig through the biosample life stages and ages
         if (result.related_datasets && result.related_datasets.length > 0) {
             result.related_datasets.forEach((dataset) => {
+                if (dataset.assay_term_name) {
+                    assays.push(dataset.assay_term_name);
+                }
                 if (dataset.replicates && dataset.replicates.length > 0) {
                     dataset.replicates.forEach((replicate) => {
                         if (replicate.library && replicate.library.biosample) {
                             const biosample = replicate.library.biosample;
                             const lifeStage = (biosample.life_stage && biosample.life_stage !== 'unknown') ? biosample.life_stage : '';
+                            if (biosample.life_stage !== 'unknown' && biosample.life_stage && biosample.age_display) {
+                                const fullStage = `${biosample.life_stage[0].toUpperCase()}${biosample.age_display.split(' ')[0]}`;
+                                fullStages.push(fullStage);
+                                ages.push(biosample.age_display.split(' ')[0]);
+                                ageUnits = biosample.age_display.split(' ')[1];
+                            }
 
-                            if (lifeStage) { lifeStages.push(lifeStage); }
-                            if (biosample.age_display) { ages.push(biosample.age_display); }
-                            if (biosample.treatments) { treatments = [...treatments, ...biosample.treatments]; }
+                            if (biosample.post_synchronization_time) {
+                                postSynchTime.push(biosample.post_synchronization_time);
+                            }
+                            if (biosample.synchronization) {
+                                synchronization = biosample.synchronization;
+                            }
+                            if (biosample.post_synchronization_time_units) {
+                                postSynchronizationTimeUnits = biosample.post_synchronization_time_units;
+                            }
+                            if (lifeStage) {
+                                lifeStages.push(lifeStage);
+                            }
+                            if (biosample.treatments && treatmentConcentration) {
+                                treatmentTerm = [...treatmentTerm, ...biosample.treatments.filter(t => t.treatment_term_name).map(t => t.treatment_term_name)];
+                                treatments = [...treatments, ...biosample.treatments.reduce((output, t) => {
+                                    if (t.amount) {
+                                        output.push(t.amount);
+                                    }
+                                    return output;
+                                }, [])];
+                                treatmentUnit = `${biosample.treatments[0].amount_units}s`;
```

Comment:
No check for value === 1?

Reply:
This was a good catch. I checked with Bonita and she says there should never be an "s" at the end for any cases.